x-pack/plugins/ingest_manager/server/services/datasource.ts

```diff
@@ -98,14 +109,61 @@ class DatasourceService {
   public async update(
     soClient: SavedObjectsClientContract,
     id: string,
-    datasource: NewDatasource
+    datasource: NewDatasource,
+    options?: { user?: AuthenticatedUser }
   ): Promise<Datasource> {
-    await soClient.update<Datasource>(SAVED_OBJECT_TYPE, id, datasource);
+    const oldDatasource = await this.get(soClient, id);
+
+    if (!oldDatasource) {
+      throw new Error('Datasource not found');
+    }
+
+    await soClient.update<Datasource>(SAVED_OBJECT_TYPE, id, {
+      ...datasource,
+      revision: oldDatasource.revision + 1,
+    });
+
+    // Bump revision of associated agent config
+    await agentConfigService.update(soClient, datasource.config_id, {}, { user: options?.user });
```

Comment:
maybe we can have a method with a more explicit name and avoid the comment  `agentConfigService.bumpRevision` ? or `agentConfigService.datasourceUpdate`?

Reply:
added `bumpRevision`